packages/common/src/logger.js

```diff
@@ -16,31 +16,53 @@
 
  'use strict';
 
-var splice = Array.prototype.splice;
-var join = Array.prototype.join;
-var makeLogger = require('log-driver');
-
-module.exports = {
-  /**
-   * Factory method that returns a new logger.
-   *
-   * @param {string=} level Log level for reporting to the console.
-   * @param {?string=} prefix to use in log messages.
-   */
-  create: function(level, prefix) {
-    var prefix_ = prefix || '';
-
-    return makeLogger({
-      levels: ['error', 'warn', 'info', 'debug', 'silly'],
-      level: level || 'error',
-      format: function() {
-        arguments[0] = arguments[0].toUpperCase();
-        splice.call(arguments, 1, 0, ':', prefix_, ':');
-        return join.call(arguments, ' ');
-      }
-    });
-  } /* create */
-
-}; /* module.exports */
+/*!
+ * @module common/logger
+ */
+
+var format = require('string-format-obj');
+var is = require('is');
+var logDriver = require('log-driver');
+
+/**
+ * Create a logger to print output to the console.
+ *
+ * @param {string=|object=} options - Configuration object. If a string, it is
+ *     treated as `options.level`.
+ * @param {string=} options.level - The minimum log level that will print to the
+ *     console. (Default: `error`)
+ * @param {string=} options.tag - A tag to use in log messages.
+ */
+function logger(options) {
+  if (is.string(options)) {
+    options = {
+      level: options
+    };
+  }
+
+  options = options || {};
+
+  return logDriver({
+    levels: [
+      'error',
+      'warn',
+      'info',
+      'debug',
+      'silly'
+    ],
+
+    level: options.level || 'error',
+
+    format: function() {
+      var args = [].slice.call(arguments);
 
+      return format('{level}{tag} {message}', {
+        level: args.shift().toUpperCase(),
+        tag: options.tag ? ':' + options.tag + ':' : '',
```

Comment:
Yes, and I did originally have it that way, but that is a bit of a micro-optimization, where it affected logical, behavioral placement more than I liked. At the time `logger()` is invoked, `tag` doesn't necessarily need to be formatted, and it's formatting there has no relevance. For now, we're only using `tag` for a single purpose, but theoretically, it could be required in multiple places throughout the function, where its original form is expected. It's only when the entire message is formatted that the tag's formatting is important.

Basically, it just came down to this for me: if someday down the road, I'm looking to change the formatting of the printed log line, I would expect to find all formatting logic to be in the `format` function.